libs/components/datetime/src/lib/modules/datepicker/datepicker-input.directive.ts

```diff
@@ -540,4 +504,39 @@ export class SkyDatepickerInputDirective
       this.adapter.setPlaceholder(this.elementRef, this.dateFormat);
     }
   }
+
+  private updateValue(value: any, isProgramatic = false): void {
+    if (this._value === value) {
+      return;
+    }
+
+    const dateValue = this.getDateValue(value);
+
+    const areDatesEqual =
+      this._value instanceof Date &&
+      dateValue &&
+      dateValue.getTime() === this._value.getTime();
+
+    const isValidDateString = this.isDateStringValid(value);
+
+    // If the string value supplied is malformed, do not set the value to its Date equivalent.
+    // (JavaScript's Date parser will convert poorly formatted dates to Date objects, such as "abc 123", which isn't ideal.)
+    if (!isValidDateString) {
+      this._value = value;
+      this.notifyUpdatedValue(isProgramatic);
+    } else if (dateValue !== this._value || !areDatesEqual) {
+      this._value = dateValue || value;
+      this.notifyUpdatedValue(isProgramatic);
+    }
+
+    if (dateValue && isValidDateString) {
+      const formattedDate = this.dateFormatter.format(
+        dateValue,
+        this.dateFormat
+      );
+      this.setInputElementValue(formattedDate);
+    } else {
+      this.setInputElementValue(value || '');
+    }
```

Comment:
☝️ Are all these changes a simple copy/paste from the original setter? Any changes that I need to focus on?

Reply:
Just a copy and paste of the original setter other than the new if statement inside `notifyUpdatedValue` for `isProgrammatic` where we set the controls value without using onChange and then the pass through of the variable to that function.